translator/src/rlp/telos_rlp_decode.rs

```diff
@@ -88,10 +89,15 @@ impl TelosTxDecodable for TxLegacy {
 
         let signed = tx.into_signed(sig);
         if buf.len() + header.payload_length != original_len {
```

Comment:
This can be negated for early return if you prefer:

```rust
        if buf.len() + header.payload_length == original_len || buf.iter().all(|&b| b == 128) {
            return Ok(signed);
        }

        error!("Transaction has trailing non-zero RLP values: {:?}", signed);
        Err(Error::ListLengthMismatch {
            expected: header.payload_length,
            got: original_len - buf.len(),
        })
```

but is the intention to return error only if len is different and there are values other than 128? If that is the padding then 👍

Reply:
No, intention is an OR but really there should never be any values in the buffer at all if the length check passes.  But your code is cleaner I think, updated the PR.

Reply:
Maybe just naming `buf.len()` as `padding_len` would make things clearer. Also, it seems that returned error is misleading.

```rust
    let padding_len = buf.len();
    if original_len - padding_len != header.payload_length {
        return Err(Error::ListLengthMismatch {
            expected: header.payload_length,
            got: original_len - padding_len,
        });
    }
    
    if padding_len > 0 && buf.iter().any(|&b| b != 128) {
        error!("Transaction has trailing non-zero RLP values: {:?}", signed);
        return Err(Error::TrailingNonZeroRLPValues);
    }
    
    Ok(signed)
}
```

Sorry for time-waster :D